### PR TITLE
🐛 byok: validate encryption class keyID if specified

### PR DESCRIPTION
**Pull request by dougm**
56a5c8a2 removed keyID validation, as 'IsValidKey' used the ListKeys API which only returns keys cached in vCenter. The 'IsValidKey' impl now uses 'QueryCryptoKeyStatus' which checks keyID against the KMS.

### CODE REVIEW

File: pkg/vmconfig/crypto/crypto_reconciler_pre.go [hunk 1]
```diff
@@ -546,6 +550,12 @@ func getCryptoKeyFromEncryptionClass(
 		return cryptoKey{}, ErrInvalidKeyProvider
 	}
 
+	if obj.Spec.KeyID != "" {
+		if ok, _ := m.IsValidKey(ctx, obj.Spec.KeyProvider, obj.Spec.KeyID); !ok {
```
Comment (dougm) [inline pkg/vmconfig/crypto/crypto_reconciler_pre.go:554]: Note that this method of key validation is not supported with the Native key provider. The Native key provider does not support directly generating keys either. So it seems specifying a `keyID` with native `keyProvider` is not a valid use-case? If it is considered a valid use-case, will need to update this to skip calling `IsValidKey` if the provider is native.

Comment (akutz) [reply pkg/vmconfig/crypto/crypto_reconciler_pre.go:554]: Earlier in development, I tracked if the provider was native. We can reintroduce that for this.

Comment (akutz) [reply pkg/vmconfig/crypto/crypto_reconciler_pre.go:554]: Earlier in development, I tracked if the provider was native. We can reintroduce that for this.

**Review by akutz — APPROVED** [review]
Thanks @dougm , looks good to me!

**Pull request merged**